app/controllers/assays_controller.rb

```diff
@@ -112,6 +111,13 @@ def create
     end
   end
 
+
+  def delete_linked_sample_types
+    return unless is_single_page_assay?
+
+    @assay.sample_type.delete
```

Comment:
you should use `.destroy` to ensure any callbacks are called. `delete `will just delete the record from the database without any checks

Reply:
Thanks, didn't know that. See commit [eed9e1c](https://github.com/seek4science/seek/pull/1571/commits/eed9e1c917c301a49193a01bb1cea3fd3bed4ad3).